### DISCUSSION TITLE
Turbo repo not building dev dependency local packages first

### DISCUSSION THREAD

**Discussion by bracesproul** [vercel/turborepo#8955] [Help]
### Summary

I have a monorepo which uses turbo repo. One of my packages has `devDependencies` on a handful of other packages. I'm unable to get it to build all these packages first (without explicitly saying one package depends on the other in `turbo.json`) unless I set the version to the latest version, then it works.

The issue is with the `langchain` package of [this repo](https://github.com/langchain-ai/langchainjs). It depends on [`@langchain/google-vertexai`](https://github.com/langchain-ai/langchainjs/tree/main/libs/langchain-google-vertexai) via a dev dep. If that dev dep version does not match the version in `@langchain/google-vertexai` `package.json` file then it does not build it first when building `langchain`

### Additional information

Repo: https://github.com/langchain-ai/langchainjs

Problematic package: https://github.com/langchain-ai/langchainjs/tree/main/langchain

It depends on `@langchain/google-vertexai` via a dev dep, however it does not build first when building `langchain`.

See this github action for logs:
https://github.com/langchain-ai/langchainjs/actions/runs/10275964080/job/28435693015?pr=6437

**Comment by anthonyshew** [comment]
>  If that dev dep version does not match the version in @langchain/google-vertexai package.json file then it does not build it first when building langchain

This is sounding like the primary issue to me. Are you saying that the `version` field in `package.json` is different from the version tag being assigned in the `package.json` of dependents? If that is the case, then the package manager is going to pull from the npm registry, rather than using your Internal Package.

It's possible I'm making too many assumptions here, but it's plausible.

---

For what it's worth, I cloned the repo, installed packages, and ran `turbo build`, and am seeing recursive `turbo` invocations. I highly recommend against doing this, as it breaks assumptions around `turbo`'s handling of your package and task graphs.

I'm also noticing that Deno is in the repository, which we don't support. Again, I don't know if either of these things are necessarily the source of this exact problem since I only have 2 minutes of familiarity with the repo, but they're making me suspicious.
